core/shared/src/main/scala/org/http4s/ServerSentEvent.scala

```diff
@@ -97,40 +91,31 @@ object ServerSentEvent {
         Pull.output1(sse) >> go(emptyBuffer, None, None, None, emptyBuffer, stream)
       }
 
-      def handleLine(
-          field: String,
-          value: String,
-          stream: Stream[F, String],
-      ): Pull[F, ServerSentEvent, Unit] =
-        field match {
-          case "" =>
-            go(dataBuffer, eventType, id, retry, commentBuffer.append(value), stream)
-          case "event" =>
-            go(dataBuffer, Some(value), id, retry, commentBuffer, stream)
-          case "data" =>
-            go(dataBuffer.append(value), eventType, id, retry, commentBuffer, stream)
-          case "id" =>
-            val newId = EventId(value)
-            go(dataBuffer, eventType, Some(newId), retry, commentBuffer, stream)
-          case "retry" =>
-            val newRetry = Try(value.toLong).toOption.orElse(retry)
-            go(dataBuffer, eventType, id, newRetry, commentBuffer, stream)
-          case _ =>
-            go(dataBuffer, eventType, id, retry, commentBuffer, stream)
-        }
-
       stream.pull.uncons1.flatMap {
         case None =>
           Pull.done
         case Some(("", stream)) =>
           dispatch(stream)
         case Some((s, stream)) =>
-          (FieldSeparator.split(s, 2): @unchecked) match {
-            case Array(field, value) =>
-              handleLine(field, value, stream)
-            case Array(line) =>
-              handleLine(line, "", stream)
-          }
+          if (s.startsWith("data:")) {
+            go(dataBuffer.append(dropPrefix(s, 5)), eventType, id, retry, commentBuffer, stream)
+          } else if (s.startsWith("id:")) {
+            go(dataBuffer, eventType, Some(EventId(dropPrefix(s, 3))), retry, commentBuffer, stream)
+          } else if (s.startsWith("event:")) {
+            go(dataBuffer, Some(dropPrefix(s, 6)), id, retry, commentBuffer, stream)
+          } else if (s.startsWith("retry:")) {
+            val newRetry = Try(dropPrefix(s, 6).toLong).toOption.orElse(retry)
+            go(dataBuffer, eventType, id, newRetry, commentBuffer, stream)
+          } else if (s.startsWith(":")) {
+            go(dataBuffer, eventType, id, retry, commentBuffer.append(dropPrefix(s, 1)), stream)
+          } else
+            s match {
```

Comment:
Translating this into a series of `case` clauses would be a bit more verbose but much easier to read, IMHO.

Reply:
updated. it does seem easier on eyes. thx